pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -446,6 +454,19 @@ pub mod pallet {
 				did: did_identifier.clone(),
 			};
 
+			// We don't care if an account is linked to a new DID (handled below by
+			// replacing the old with the new), but we care that if the unique linking flag
+			// is on, the DID can have a single account linked.
+			let is_did_already_linked = ConnectedAccounts::<T, I>::iter_key_prefix(&did_identifier)
+				.next()
+				.is_some();
+			let is_unique_flag_enabled = <T as Config<I>>::UniqueLinkingEnabled::get();
+			// Either flag not enabled, or if it is, the DID is not already linked.
+			ensure!(
+				!is_unique_flag_enabled || !is_did_already_linked,
+				Error::<T, I>::LinkExisting
+			);
```

Comment:
Is this right? We only need to check  `ConnectedAccounts`, if `is_unique_flag_enabaled` is true, or not?

Reply:
This reads as "Either the unique flag is not on (in which case we don't care), or (if it's on) make sure the DID is not already linked". It's equivalent to `(is_unique_flag_enabled && !is_did_already_linked)`.

Reply:
Yes, I was more considering the performance. If the flag is enabled, then we don't need to check if the did has already an association. But this is probably a bit nitpicky.

Reply:
Yeah good point, I'll change that.

Reply:
https://github.com/KILTprotocol/kilt-node/pull/782/commits/f853e2bf22ad70fbb88e8b0c37dcf4761195581e